doc/architectural/goto-harness.md

```diff
@@ -63,34 +80,66 @@ The options we pass to `goto-harness` are:
 * `harness-type` is the type of harness to generate (`call-function` is the
   function-call harness generator)
 * `function` is the name of the function that gets instrumented
-* then we pass the input GOTO-binary and a name for the output GOTO binary.
+* then we pass the input GOTO-binary and a name for the output C file.
 
-What comes of this is a GOTO binary roughly corresponding to the following C
-pseudocode:
+What comes out of this is a C file that looks like this:
 
 ```C
-// modified_goto_binary.c
-
-#include <assert.h>
-
-int function_to_test(int some_argument)
+// function_to_test
+// file main.c line 3
+signed int function_to_test(signed int some_argument);
+// harness
+// 
+void harness(void);
+// type_constructor_int
+// 
+void type_constructor_int(signed int depth_int, signed int *result_int);
+
+// __GOTO_HARNESS::max_depth
+// file __GOTO_HARNESSharness.c
+signed int max_depth=2;
+// __GOTO_HARNESS::min_depth
+// file __GOTO_HARNESSharness.c
+signed int min_depth=1;
+
+// harness
+// 
+void harness(void)
 {
-  assert(some_argument == 0);
-  return some_argument;
+  signed int some_argument;
+  type_constructor_int(0, &some_argument);
+  function_to_test(some_argument);
 }
 
-void harness(void)
+// type_constructor_int
+// 
+void type_constructor_int(signed int depth_int, signed int *result_int)
 {
-  int argument = nondet_int();
-  function_to_test(argument);
+  signed int nondet;
+  *result_int = nondet;
 }
 ```
 
-You can pass that modified GOTO binary to `cbmc` and select `harness` as the
-entry function for analysis, or further modify it with `goto-harness` or
-`goto-instrument` or other tools.
+After you have generated the harness file, you have two choices. The first
+one is to pass it along with the original file containing the function to
+be instrumented to CBMC for analysis, something that would look like this:
```

Comment:
Is instrumented the right word here?

Reply:
I changed it to something that's more appropriate for the context, I feel.